cores/arduino/CDCSerialClass.cpp

```diff
@@ -36,6 +36,33 @@ extern void CDCSerial_Handler(void);
 extern void serialEventRun1(void) __attribute__((weak));
 extern void serialEvent1(void) __attribute__((weak));
 
+static void cdc_mbox_isr(CurieMailboxMsg msg)
```

Comment:
This function looks really complex for such a simple task. Can't you just do something like this;

```
unsigned char *ptr = (unsigned char*) msg.data;

/* Continue until 16 chars, or until NULL is seen */
for (unsigned i = 0; i < 16 && ptr[i]; ++i) {
    /* Assuming we have some kind of 'put' method for ring buffer... */
    ringbuffer_put(ptr[i]);
}
```